query-node/schema.graphql

```diff
@@ -8,36 +13,149 @@ type Block @entity {
+  PAID
+  SCREENING
+  GENESIS
+}
+
 "Stored information about a registered user"
-type Member @entity {
+type Membership @entity {
   "MemberId: runtime identifier for a user"
   id: ID!
 
   "The unique handle chosen by member"
-  handle: String @unique @fulltext(query: "membersByHandle")
+  handle: String! @unique @fulltext(query: "membersByHandle")
 
   "A Url to member's Avatar image"
   avatarUri: String
 
   "Short text chosen by member to share information about themselves"
   about: String
 
-  "Blocknumber when member was registered"
-  registeredAtBlock: Int!
-
   "Member's controller account id"
   controllerAccount: Bytes!
 
   "Member's root account id"
   rootAccount: Bytes!
 
+  "Blocknumber when member was registered"
+  registeredAtBlock: Block!
+
+  "Timestamp when member was registered"
+  registeredAtTime: DateTime!
+
+  "How the member was registered"
+  entry: MembershipEntryMethod!
+
+  "The type of subscription the member has purchased if any."
+  subscription: BigInt
+}
+
+"Category of media channel"
+type ChannelCategory @entity {
+  id: ID!
+
+  name: String
+}
+
+"Storage asset"
+union Asset = AssetUrl | AssetStorage
+
+"Asset stored at an external source"
+type AssetUrl @variant {
+  id: ID!
+
+  "The http url pointing to the media"
+  url: String!
+}
+
+"Asset was never fully uploaded."
+type AssetNeverProvided @variant {
+  happenedIn: Block!
+}
+
+"Asset was deleted and is no longer available."
+type AssetDeleted @variant {
+  happenedIn: Block!
+}
+
+"Status of an asset upload"
+type AssetUploadStatus @variant {
+  """
+  Data object in upload life-cycle.
+  If this is deleted, then set oldDataObject in its place if it is set and not rejected, otherwise union goes to Deleted.
+  """
+  dataObject: AssetDataObject!
+
+  """
+  Possible prior data object which was in some stage of upload life-cycle when new one was initiated.
+  If accepted, then apps may chose to use old in place of new before it is accepted.
+  If this is deleted, then set to null.
+  """
+  oldDataObject: AssetDataObject
+
   happenedIn: Block!
 }
 
+union AssetStorageUploadStatus = AssetNeverProvided | AssetDeleted | AssetUploadStatus
+
+type AssetStorage @variant {
+  id: ID!
+
+  "Upload to content directory status"
+  uploadStatus: AssetStorageUploadStatus!
+}
+
+"The decision of the storage provider when it acts as liaison"
+enum LiaisonJudgement {
+  "Content awaits for a judgment"
+  PENDING,
+
+  "Content accepted"
+  ACCEPTED,
+
+  "Content rejected"
+  REJECTED,
+}
+
+"Manages content ids, type and storage provider decision about it"
+type AssetDataObject @entity {
+  "Content owner"
+  owner: AssetOwner!
+
+  "Content added at"
+  addedAt: Block!
+
+  "Content type id"
+  type_id: Int!
+
+  "Content size in bytes"
+  size: BigInt!
+
+  "Storage provider id of the liaison"
+  liaisonId: BigInt!
+
+  "Storage provider as liaison judgment"
+  liaisonJudgement: LiaisonJudgement!
+
+  "IPFS content id"
+  ipfsContentId: String!
```

Comment:
Just to make sure, this is the old `dataObjectId`?

Reply:
I assume so since it's only `id` in `storage`'s `DataObjectInternal`.

Reply:
IPFS content id is the IPFS hash of the content.

Reply:
Maybe rename it to avoid this confusion ?

Reply:
If this isn't the `dataObjectId`, then isn't it missing from `AssetDataObject`?

Reply:
The discussed id was really missing. It is now called `joystreamContentId` to distinguish it from IPFS id.